lib/public_suffix.ex

```diff
@@ -77,6 +127,13 @@ defmodule PublicSuffix do
          labels -> Enum.join(labels, ".")
        end
   end
+  defp parse_domain(domain, options, _, :prevailing_rule) do
+    domain
+    |> String.downcase
+    |> String.split(".")
+    |> find_prevailing_rule(options)
+    |> Enum.join(".")
+  end
```

Comment:
I think it's confusing to overload `parse_domain` to do entirely separate things based on the flag provided as the last argument.  Instead, WDYT about removing the last arg to `parse_domain` and inlining this logic in `prevailing_rule/2`?